crypto/mem.c

```diff
@@ -294,6 +294,9 @@ void *CRYPTO_clear_realloc(void *str, size_t old_len, size_t num,
 
 void CRYPTO_free(void *str, const char *file, int line)
 {
+    if (str == NULL)
```

Comment:
This is not necessary since the C runtime library requires that ```free(NULL)``` do the obvious right thing.